controllers/watchers/target_pod.go

```diff
@@ -0,0 +1,563 @@
+	}
+
+	// Sort by last timestamp
+	sort.Slice(eventList.Items, func(i, j int) bool {
+		return eventList.Items[i].LastTimestamp.After(eventList.Items[j].LastTimestamp.Time)
+	})
+
+	// Keep events sent during the disruption only, no need to filter events coming from the disruption itself
+	if kind != "Disruption" {
+		for i, event := range eventList.Items {
+			if event.Type == v1.EventTypeWarning && event.Reason == v1beta1.EventDisrupted || event.LastTimestamp.Time.Before(disruptionStateTime) {
+				if i == 0 {
+					return []v1.Event{}, nil
+				}
+
+				return eventList.Items[:(i - 1)], nil
+			}
+		}
+	}
+
+	return eventList.Items, nil
+}
+
+func getContainerState(containerStatus v1.ContainerStatus) (string, string) {
+	var state, reason string
+
+	switch {
+	case containerStatus.State.Running != nil:
+		state = "Running"
+	case containerStatus.State.Waiting != nil:
+		state = "Waiting"
+		reason = containerStatus.State.Waiting.Reason
+	case containerStatus.State.Terminated != nil:
+		state = "Terminated"
+		reason = containerStatus.State.Terminated.Reason
+	}
+
+	return state, reason
+}
+
+// https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/events/event.go
+// In case of new events sent from kubelet, we can determine any error event in the node to propagate it
+func (d DisruptionTargetHandler) findNotifiableEvents(eventsToSend map[string]bool, eventsFromTarget []v1.Event, recoverTimestamp *time.Time, targetName string) (map[string]bool, bool) {
+	cannotRecoverYet := true
+
+	for _, event := range eventsFromTarget {
+		if event.Source.Component == v1beta1.SourceDisruptionComponent {
+			// if the target has not sent any warnings, we can't recover it as there is nothing to recover
+			if v1beta1.IsTargetEvent(event) && event.Type == v1.EventTypeWarning {
+				cannotRecoverYet = false
+			}
+
+			break
+		}
+	}
+
+	// ordered by last timestamp
+	for _, event := range eventsFromTarget {
+		// We stop at the last event sent by us
+		if event.Source.Component == v1beta1.SourceDisruptionComponent || (recoverTimestamp != nil && recoverTimestamp.After(event.LastTimestamp.Time)) {
+			break
+		}
+
+		// if warning event has been sent after target recovering
+		switch event.InvolvedObject.Kind {
+		case "Pod":
+			switch {
+			case event.Type == v1.EventTypeWarning:
+				if event.Reason == "Unhealthy" || event.Reason == "ProbeWarning" {
+					lowerCasedMessage := strings.ToLower(event.Message)
+
+					switch {
+					case strings.Contains(lowerCasedMessage, "liveness probe"):
+						eventsToSend[v1beta1.EventTargetLivenessProbeChange] = true
+					case strings.Contains(lowerCasedMessage, "readiness probe"):
+						// If the object of the disruption is in the list of targets, it means it has been injected.
+						// The readiness probe is failing during the injection
+						if d.Disruption.Status.HasTarget(event.InvolvedObject.Name) {
+							eventsToSend[v1beta1.EventTargetReadinessProbeChangeDuringDisruption] = true
+						} else {
```

Comment:
Is this `else` guaranteed to be true? Could it be the case that we no longer have the Target in our Targets list, but we're still getting events about it from the watcher? If this watcher is for our targets, how are we getting an event about this pod but it isn't in our Targets list?

Reply:
> Could it be the case that we no longer have the Target in our Targets list, but we're still getting events about it from the watcher?
To be honest, I don't know... I just move the **DisruptionTargetHandler** form the cache_handler.go...

Reply:
If it isn't a change in behavior, it's fine if its wrong.